src/dodona/util/Interactive.java

```diff
@@ -47,97 +48,142 @@ private Interactive(final Class<?> cls) {
     @Override
     public Statement apply(final Statement base, final Description description) {
         // Ugly hack to use existing Rules as Rules.
-        return this.feedback.apply(this.stdin.apply(this.stdout.apply(
-            base, description),
+        return this.feedback.apply(this.diff.apply(this.stdin.apply(
+            this.stdout.apply(base, description),
+            description),
             description),
             description);
     }
     
     /**
+     * Asserts that the contents of stdout match the expected value.
+     *
+     * @param expected the expected output
+     * @return fluent
+     */
+    public Interactive assertOutput(final String expected) {
+        // Remove the output from the feedback stream if any.
+        if (this.outputMessageId > -1) {
+            this.feedback.remove(this.outputMessageId);
+            this.outputMessageId = -1;
+        }
+        // Perform the comparison.
+        this.diff.compare(expected, this.output());
+        return this;
+    }
+    
+    /**
+     * Allow messages to be appended in TestCarryingThrowable
      * Calls the .main()-method of the class under test.
      *
+     * @param args commandline arguments
      * @return fluent
      * @throws Throwable exception thrown by the program
      */
-    public Interactive callMain() throws Throwable {
+    public Interactive callMain(final String[] args) throws Throwable {
```

Comment:
Use `public Interactive callMain(final String... args)` and you can remove the extra constructor below. And you can call it as `callMain("arst", "neio")` instead of `callMain(new String[]{...})`.